src/python/ensembl/io/genomio/gff3/id_allocator.py

```diff
@@ -173,32 +173,40 @@ def normalize_gene_id(self, gene: SeqFeature) -> str:
         prefixes = ["gene-", "gene:"]
         new_gene_id = StableIDAllocator.remove_prefix(gene.id, prefixes)
 
+        is_valid = False
+        # Special case for RefSeq: only valid Gene IDs are LOC*
+        if refseq:
+            if new_gene_id.startswith("LOC"):
+                is_valid = True
+        else:
+            is_valid = self.is_valid(new_gene_id)
+
+        if is_valid:
+            return new_gene_id
+
         # In case the normalized gene ID is not valid, use the GeneID
-        if not self.is_valid(new_gene_id):
-            logging.debug(f"Gene ID is not valid: {new_gene_id}")
-            qual = gene.qualifiers
-            if "Dbxref" in qual:
-                for xref in qual["Dbxref"]:
-                    (db, value) = xref.split(":")
-                    if db != "GeneID":
-                        continue
-                    new_gene_id_base = f"{db}_{value}"
-                    new_gene_id = new_gene_id_base
-                    number = 1
-                    while new_gene_id in self._loaded_ids:
-                        number += 1
-                        new_gene_id = f"{new_gene_id_base}_{number}"
-                        if number > 10:
-                            raise InvalidStableID(f"Duplicate ID {new_gene_id_base} (up to {new_gene_id})")
-                    self._loaded_ids.add(new_gene_id)
-                    logging.debug(f"Using GeneID {new_gene_id} for stable_id instead of {gene.id}")
-                    return new_gene_id
-
-            # Make a new stable_id
-            if self.make_missing_stable_ids:
-                new_gene_id = self.generate_gene_id()
-                logging.debug(f"New ID: {new_gene_id} -> {new_gene_id}")
+        logging.debug(f"Gene ID is not valid: {new_gene_id}")
+        qual = gene.qualifiers
+        if "Dbxref" in qual:
+            for xref in qual["Dbxref"]:
+                (db, value) = xref.split(":")
+                if db != "GeneID":
+                    continue
+                new_gene_id_base = f"{db}_{value}"
+                new_gene_id = new_gene_id_base
+                number = 1
+                while new_gene_id in self._loaded_ids:
+                    number += 1
+                    new_gene_id = f"{new_gene_id_base}_{number}"
+                    if number > 10:
```

Comment:
why do we raise a error only when the number exceeds 10, why not when we find more than 1 unique id?

Reply:
Some genomes use the same GeneID for several different genes in the same annotation, so this is to allow some room for those.